modules/citrus-http/src/test/resources/com/consol/citrus/http/integration/HttpGzipEncodedIT.xml

```diff
@@ -85,7 +85,7 @@
           <http:headers content-type="text/html">
             <http:header name="Content-Encoding" value="gzip"/>
           </http:headers>
-          <http:body>
+          <http:body type="gzip">
```

Comment:
I think the `type` declaration is something we should get rid of n the long term. I see that this is not an option as part of this change but we should consider this for a future major release. And yes, I see that this is a shitload of work. :sweat_smile:

Reply:
Now as I think about it myself, I don't know whether this is possible at all. It's important to keep in mind, that we do not only handle plaintext payload. If we've resource files, we'd have to specify the content type of the resource file anyways as we don't know if it's a plain text content or a gzip resource with prepared test data.